django/core/management/commands/dumpdata.py

```diff
@@ -6,6 +8,38 @@
 from django.core.management.utils import parse_apps_and_model_labels
 from django.db import DEFAULT_DB_ALIAS, router
 
+try:
+    import bz2
+    HAS_BZ2 = True
+except ImportError:
+    HAS_BZ2 = False
+
+try:
+    import lzma
+    HAS_LZMA = True
+except ImportError:
+    HAS_LZMA = False
+
+
+def get_open_method(filename, mode='wt'):
+    """Return open method an mode based on the filename suffix."""
+    try:
+        filename_suffix = pathlib.PurePath(filename).suffix
+    except TypeError:
+        filename_suffix = ''
```

Comment:
I don't think we need to use `pathlib` here. Just use `str.rsplit()` as `loaddata` does.

Reply:
Pathlib already extract suffix from a filename, so why reinvent the wheel ?
What do you think are the contraindications to using Pathlib?

Reply:
It just seems overkill when you are checking the string ends with a particular value.

Reply:
I updated this line with a `rsplit`